### PR TITLE
initial target support

### PR DESCRIPTION
**Pull request by tpolecat**
This is incomplete (needs tests and fixes for sure) but I'd like to merge it so @swalker2m can have a working build. It's huge, sorry. Recommend crossing fingers.

### CODE REVIEW

**Comment by tpolecat** [comment]
Next up after this is @armanbilge's PR that updates the build, I think.

**Review by swalker2m — COMMENTED** [review]
I just skimmed through it but it looks great.  The target stuff is the most complicated part of the schema so far so you've got a good bit done.  I noticed there are a lot of "unused" warnings in the compile.  Are we not failing the build for those for now?

**Comment by swalker2m** [comment]
Should (can?) we switch to Scala 3?  Not that I'm a keener or anything.

File: build.sbt [hunk 1]
```diff
@@ -4,15 +4,15 @@ val clueVersion                = "0.20.2"
...
 val disciplineMunitVersion     = "1.0.9"
 val flywayVersion              = "7.11.4"
-val grackleVersion             = "0.1.16+17-6f247fe9+20220407-1243-SNAPSHOT"
+val grackleVersion             = "0.2.0" // "0.1.16+17-6f247fe9+20220407-1243-SNAPSHOT"
 val http4sVersion              = "0.23.6"
 val jwtVersion                 = "5.0.0"
 val log4catsVersion            = "2.1.1"
-val lucumaCoreVersion          = "0.14.3"
-val lucumaGraphQLRoutesVersion = "0.2.0-3-bec8835-20220407T175637Z-SNAPSHOT"
+val lucumaCoreVersion          = "0.2.0" //"0.14.3"
```
[the remark this answers is not in the document]
Comment (tpolecat) [reply build.sbt:11]: Oh, that was a mistake. I'm not using much from the core model so I didn't notice.

**Comment by tpolecat** [comment]
Not failing from unused warnings yet but we could be. Need to merge Arman's PR and see how things look.

**Pull request merged**